README.md

```diff
@@ -1,21 +1,77 @@
 # golf-course
 A list of common Solidity optimization tips and myths.
 
-## Tips
+# Tips
 
-### Right Shift Instead of Dividing By 2
+- - - -
+### 1.  When dividing by two, use `>> 1` instead of `/ 2` ###
+
+```solidity
+/// 🤦 Unoptimized (gas: 1401)
+uint256 two = 4 / 2;
+
+/// 🚀 Optimized (gas: 1372)
+uint256 two = 4 >> 1;
+```
 
 The `SHR` opcode is 3 gas cheaper than `DIV` and also bypasses Solidity's division by 0 prevention overhead.
+  - [Gas Usage]()
+  - [Full Example](https://github.com/Rari-Capital/golf-course/blob/fc1882bacfec50787d9e9435d59fed4a9091fb21/src/GolfCourse.t.sol#L15)
+
+
+- - - -
+
+### 2.  Use `>=` and `<=` instead of `>` and `<` ###
+```solidity
+/// 🤦 Unoptimized (gas: 1401)
+bool positive = 1 > 0;
+
+/// 🚀 Optimized (gas: 1337)
+bool positive = 1 >= 1;
+```
+
+`if (x >= y) {}` is algebraically equivalent to !(x < y) which gets compiled to `LT .. JUMPI` which is more efficient than `x > y` which contains an extra `ISZERO`
+
+- [Gas Usage]()
+- [Full Example](https://github.com/Rari-Capital/golf-course/blob/fc1882bacfec50787d9e9435d59fed4a9091fb21/src/GolfCourse.t.sol#L23)
+
+- - - -
+
+### 3. Using `!=` is usually cheaper than `>` or `<` ###
+```solidity
+/// 🤦 Unoptimized (gas: 1423)
```

Comment:
haha, I couldn't get this "tip" to work ... maybe it's a myth?

Reply:
i think cuz of

- function overhead in dapptools tests
- we're using constants and not dynamic values so the compiler can optimize at compile time

Reply:
Tried it with dynamic values for both sides of the equation.  Also tried different values (like zero and non zero)  No matter what I do, I was unable to repro gas savings.  Taking this one off the list as well althought I'd be happy to be proved wrong on any of these.

Also, maybe because I'm using pragma 0.8.11???

Reply:
Actually I figured it out.  It only saves gas within a `require()`